### PR TITLE
Set Guild Config action

### PR DESCRIPTION
**Pull request by MiltonTulli**
# Description
Include new action in ActionsBuilder to change guild config. All fields by default will have current guild config values. Infoline component will display only updated fields with prev-> new values. 

Closes https://github.com/DXgovernance/DAVI/issues/313

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

https://user-images.githubusercontent.com/25844967/200446024-646e3b86-5d87-462a-a9df-f7fd8dca4f10.mov

Screen Shot 2022-11-07 at 21 43 38
Screen Shot 2022-11-07 at 21 43 24

### CODE REVIEW

File: src/components/ActionsBuilder/SupportedActions/SetGuildConfig/SetGuildConfigEditor.test.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      lockTime: mockBigNumber,
+      voteGas: mockBigNumber,
+      maxGasPrice: mockBigNumber,
+      votingPowerPercentageForProposalExecution: mockBigNumber,
+      votingPowerPercentageForProposalCreation: mockBigNumber,
+      minimumMembersForProposalCreation: mockBigNumber,
+      minimumTokensLockedForProposalCreation: mockBigNumber,
+      votingPowerForProposalExecution: mockBigNumber,
+      token: '0x30ii2jnn32ru8238r293ru2389ru2938ru239r23r23',
+    },
```
Comment (dcrescimbeni) [inline src/components/ActionsBuilder/SupportedActions/SetGuildConfig/SetGuildConfigEditor.test.tsx:28]: Should we move this to the `fixtures` file?

**Review by dcrescimbeni — COMMENTED** [review]
Awesome! This has a lot of good patterns that we should use for the other forms.

Left some minor comments, but looks good :+1:

File: src/components/ActionsBuilder/SupportedActions/SetGuildConfig/utils.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+type UpdatedValuesReturn = {
+  [key in ControlField]: {
+    newValue: BigNumber;
+    currentValue: BigNumber;
+  };
+};
+export const getUpdatedValues = (
+  current: SetGuildConfigFields,
+  modifyed: SetGuildConfigFields
```
[the remark this answers is not in the document]
Comment (MiltonTulli) [reply src/components/ActionsBuilder/SupportedActions/SetGuildConfig/utils.ts]: 😁

File: src/components/ActionsBuilder/SupportedActions/SetGuildConfig/SetGuildConfigInfoLine.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  InfoLineTitle,
+  InfoLineConfigItem,
+} from './SetGuildConfigEditor.styled';
+import { Flex } from 'components/primitives/Layout/Flex';
+
+const getDisplayValue = (value: BigNumber, type: FieldType) => {
+  if (type === FieldType.number) return bn(value).toNumber();
+  if (type === FieldType.percentage) return `${bn(value).toNumber()}%`;
+  if (type === FieldType.duration)
+    return duration(bn(value).toNumber(), 'seconds').humanize();
```
[the remark this answers is not in the document]
Comment (MiltonTulli) [reply src/components/ActionsBuilder/SupportedActions/SetGuildConfig/SetGuildConfigInfoLine.tsx]: Yep `.humanize()` is kind of random but is safe too. 
I did a refactor to use your logic from `useDuration` and I am using it on the infoline component to display values. I didn't touch any of the logic (suggest a refactor), just extracted it to be able to reuse it.

File: src/components/ActionsBuilder/SupportedActions/SetGuildConfig/SetGuildConfigEditor.test.tsx (hunk 1 above)
Comment (MiltonTulli) [reply src/components/ActionsBuilder/SupportedActions/SetGuildConfig/SetGuildConfigEditor.test.tsx:28]: I don't think we need to use fixtures for mocks. Fixtures are more for testing diff outcomes I guess. Also, jest doesn't like out-of-scope content for some reason.

**Comment by rossneilson** [comment]
I think we have some maths issues here, should be divided by 100
Screenshot 2022-11-10 at 12 09 02

But this is awesome!

**Comment by rossneilson** [comment]
Also could we show the plain language version of params instead of camelcase in the action line component?

**Comment by dcrescimbeni** [comment]
LGTM :fire:

**Comment by dcrescimbeni** [comment]
> I think we have some maths issues here, should be divided by 100 Screenshot 2022-11-10 at 12 09 02
> 
> But this is awesome!

I'm seeing the correct values in the percentages:

image

Tried the three guilds and all are showing OK.

**Comment by rossneilson** [comment]
> > I think we have some maths issues here, should be divided by 100 Screenshot 2022-11-10 at 12 09 02
> > But this is awesome!
> 
> I'm seeing the correct values in the percentages:
> 
> image
> 
> Tried the three guilds and all are showing OK.

Try looking at gnosis guilds or mainnet

**Comment by MiltonTulli** [comment]
> > > I think we have some maths issues here, should be divided by 100 Screenshot 2022-11-10 at 12 09 02
> > > But this is awesome!
> > 
> > 
> > I'm seeing the correct values in the percentages:
> > image
> > Tried the three guilds and all are showing OK.
> 
> Try looking at gnosis guilds or mainnet

Thanks for testing! I was assuming that whatever contract return was the actual percentage (since the key is actually named "percentage" =) ). After review the contracts. 

These are the values I am getting from guild configs: 

/**
 *
 * - Localhost:
 *    - DxdGuild:
 *          votingPowerPercentageForProposalCreation:"1"
 *          votingPowerPercentageForProposalExecution:"30"
 *    - SwprGuild:
 *          votingPowerPercentageForProposalCreation:"5"
 *          votingPowerPercentageForProposalExecution:"30"
 *    - RepGuild:
 *          votingPowerPercentageForProposalCreation:"5"
 *          votingPowerPercentageForProposalExecution:"50"
 *
 * - Mainnet:
 *    - Devops Guild:
 *          votingPowerPercentageForProposalCreation:"500"
 *          votingPowerPercentageForProposalExecution:"5000"
 *
 *
 * - Gnosis:
 *    - Devops Guild:
 *          votingPowerPercentageForProposalCreation: "500",
 *          votingPowerPercentageForProposalExecution:"5000"
 *     - Devops Treasury:
 *          votingPowerPercentageForProposalCreation: "500",
 *          votingPowerPercentageForProposalExecution:"5000"
 *
 */
 
 I will refactor this to display correct values (where 100 == =1% and  2500 === 25% according to the contracts https://github.com/DXgovernance/dxdao-contracts/blob/develop/contracts/erc20guild/BaseERC20Guild.sol#L75)

**Comment by MiltonTulli** [comment]
> I think we have some maths issues here, should be divided by 100 Screenshot 2022-11-10 at 12 09 02
> 
> But this is awesome!

@rossneilson fixed percentage issue and displayed the label (instead of field name) on infoline component. Also I think it would be nice to have these issue-comments as review comments to keep track in a single thread and being able to resolve these issues as we go. What do you think?

**Comment by dcrescimbeni** [comment]
LGTM :100:

**Comment by Filipv95** [comment]
Hey, @MiltonTulli tested this PR as well, everything seem to be working as expected, great job!

**Comment by dcrescimbeni** [comment]
Everything looks good.

The validation is :ok_hand:

**Comment by rossneilson** [comment]
Screenshot 2022-11-18 at 17 51 15

Looks great!!!!

**Review by rossneilson — APPROVED** [review]
Amazing work, just that small verification bug but we can make that a new issue

**Pull request merged**